tests/std/tests/P0220R1_optional/test.cpp

```diff
@@ -7382,6 +7382,47 @@ namespace msvc {
             (void) t;
         }
     } // namespace vso614907
+
+    namespace gh2458 {
+        // optional<const meow> o = std::move(lvalue_optional_const_meow);
+        // was moving from the contained const object
+        enum class action { none, copy, move };
+        action last = action::none;
+
+        struct Copyable {
+            Copyable() = default;
+            Copyable(const Copyable&) {
+                last = action::copy;
+            }
+            Copyable(Copyable&&) {
+                last = action::move;
+            }
+        };
+
+        struct ConstMovable {
+            ConstMovable() = default;
+            ConstMovable(const ConstMovable&) {
+                last = action::copy;
+            }
+            ConstMovable(const ConstMovable&&) {
+                last = action::move;
+            }
+        };
+
+        template <class T, action result>

```

Comment:
Style: We conventionally capitalize template parameters, so this should be `Result`.